vpn/src/main/java/com/duckduckgo/mobile/android/vpn/bugreport/NetworkTypeCollector.kt

```diff
@@ -119,9 +121,20 @@ class NetworkTypeCollector @Inject constructor(
                 )
                 true
             } else {
-                Timber.v("Private DNS support is disabled or not set")
+                Timber.v("Private DNS disabled")
                 false
             }
+
+            // Check if VPN reconfiguration is needed
+            if (appTpFeatureConfig.isEnabled(AppTpSetting.PrivateDnsSupport) && vpnPreferences.isPrivateDnsEnabled != privateDns) {
```

Comment:
does this logic belong to the collector? it's not like the other network callbacks that actually change values used in collector.

Reply:
I just saw that [PrivateDnsSettingMonitor.kt](https://github.com/duckduckgo/Android/pull/1905/files#diff-0edd70d3143c981c20c1cc0af74c05f06dffd81c681502bbb7ab1f35023ccf34) is deleted. Why not leave it as it was with the added logic?

Reply:
Yeah, the `NetworkTypeCollector` is probably not the ideal place, is prob mainly semantics around the name of the class. But I think it was better to move it here and delete the private dns setting monitor because it also simplifies the logic around the shared prefs.